client/js/stores/snippet-store.js

```diff
@@ -38,6 +38,40 @@ class SnippetStore {
     this.preventDefault();
   }
 
+  countSnippetAverage(data) {
+    var snippetId = data.snippetId;
+    var userRating = data.userRate;
+    var avg = data.avarage;
+    var snippetsAverage = this.state.snippetsAvg;
+
+    snippetsAverage[snippetId] = avg;
```

Comment:
calculating happens in component (rating.jsx) and there's no re-calculating (except for getting avg from database on rating creation). snippet doesn't have avg as an attribute (?)     only all ratings. so we need to keep  it separately from array of ratings from current user